authors/apps/articles/views.py

```diff
@@ -0,0 +1,87 @@
+from django.shortcuts import get_object_or_404
+
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework import viewsets
+
+from .models import Article
+from .serializers import ArticleSerializer
+from .renderers import ArticleJsonRenderer
+
+
+class ArticleViewSet(viewsets.ViewSet):
+    """
+    Example empty viewset demonstrating the standard
+    actions that will be handled by a router class.
+
+    If you're using format suffixes, make sure to also include
+    the `format=None` keyword argument for each action.
+    """
+    queryset = Article.objects.all()
+    serializer_class = ArticleSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    renderer_classes = (ArticleJsonRenderer,)
+
+    def list(self, request):
+        queryset = Article.objects.all()
+        serializer = ArticleSerializer(queryset, many=True)
+        return Response({"Articles": serializer.data})
+
+    def create(self, request):
+        """create an article"""
+        article = request.data
+        serializer = self.serializer_class(
+            data=article, context={'request': request})
+        serializer.is_valid(raise_exception=True)
+        serializer.save(author=request.user)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def retrieve(self, request, pk=None):
+        """
+        Returns article with the given slug if exists
+        or returns an exception if no article with slug exists
+        """
+        queryset = Article.objects.all()
+        article = get_object_or_404(queryset, pk=pk)
+        serializer = ArticleSerializer(article)
+        return Response(serializer.data)
+
+    def update(self, request, pk=None):
+        """
+        Update an article
+        """
+        queryset = Article.objects.all()
+        article = get_object_or_404(queryset, pk=pk)
+        article_data = request.data.get('article', {})
+        serializer = self.serializer_class(
+            data=article_data, partial=True)
```

Comment:
This should have the `partial=False` and the `partial_update` should have the `partial=True`.

Reply:
Noted